educator/src/Dscp/DB/SQLite/Instances.hs

```diff
@@ -137,11 +149,87 @@ instance (HasSqlValueSyntax (Sql92ExpressionValueSyntax syntax) (TYPE), \
           IsSql92ExpressionSyntax syntax) => \
          HasSqlEqualityCheck syntax (TYPE)
 
+GenHasSqlEqualityCheck(Hash a)
 GenHasSqlEqualityCheck(Address)
 GenHasSqlEqualityCheck(Course)
+GenHasSqlEqualityCheck(AssignmentType)
 GenHasSqlEqualityCheck(Subject)
 GenHasSqlEqualityCheck(Grade)
 GenHasSqlEqualityCheck(TxBlockIdx)
 GenHasSqlEqualityCheck(BlockIdx)
-GenHasSqlEqualityCheck(Hash a)
-GenHasSqlEqualityCheck(AssignmentType)
+GenHasSqlEqualityCheck(SubmissionWitness)
+GenHasSqlEqualityCheck(MerkleSignature a)
+GenHasSqlEqualityCheck(EmptyMerkleTree a)
+GenHasSqlEqualityCheck(ATGDelta)
+
+----------------------------------------------------------------------------
+-- Backend instances
+----------------------------------------------------------------------------
+
+type TypeHasBackendInstances cmd be a =
+    ( FromBackendRow be a
+    , BackendFromField be a
+    , Beam.HasSqlValueSyntax
+        (Beam.Sql92ExpressionValueSyntax
+            (Beam.Sql92SelectTableExpressionSyntax
+              (Beam.Sql92SelectSelectTableSyntax (Beam.Sql92SelectSyntax cmd))))
+      a
+    , Beam.HasSqlValueSyntax
+        (Beam.Sql92ExpressionValueSyntax
+            (Beam.Sql92InsertValuesExpressionSyntax
+              (Beam.Sql92InsertValuesSyntax (Beam.Sql92InsertSyntax cmd))))
+      a
+    , Beam.HasSqlEqualityCheck
+      (Beam.Sql92InsertValuesExpressionSyntax
+          (Beam.Sql92InsertValuesSyntax
+            (Beam.Sql92InsertSyntax cmd)))
+      a
+    )
+
+type BackendHasInstances cmd be =
+    ( Beam.HasQBuilder (Beam.Sql92SelectSyntax cmd)
+    , Beam.IsSql92Syntax cmd
+    , TypeHasBackendInstances cmd be Int
+    , TypeHasBackendInstances cmd be Text
+    , TypeHasBackendInstances cmd be Bool
+    , TypeHasBackendInstances cmd be Address
+    , TypeHasBackendInstances cmd be Course
+    , TypeHasBackendInstances cmd be Subject
+    , TypeHasBackendInstances cmd be Grade
+    , TypeHasBackendInstances cmd be TxBlockIdx
+    , TypeHasBackendInstances cmd be BlockIdx
+    , TypeHasBackendInstances cmd be AssignmentType
+    , TypeHasBackendInstances cmd be SubmissionWitness
+    , TypeHasBackendInstances cmd be UTCTime
+    , TypeHasBackendInstances cmd be ATGDelta
+    , TypeHasBackendInstances cmd be (Hash LByteString)
+    , TypeHasBackendInstances cmd be (Hash Assignment)
+    , TypeHasBackendInstances cmd be (Hash Submission)
+    , TypeHasBackendInstances cmd be (Hash PrivateTx)
+    , TypeHasBackendInstances cmd be (Hash PrivateBlockHeader)
+    , TypeHasBackendInstances cmd be (MerkleSignature PrivateTx)
+    , TypeHasBackendInstances cmd be (EmptyMerkleTree PrivateTx)
```

Comment:
Nope, unfortunately, in particular because the main constraint in it `FromBackendRow` is a type family (`* -> Constraint`), without knowing the exact backend we cannot deduce it.